iceoryx_hoofs/include/iceoryx_hoofs/cxx/string.hpp

```diff
@@ -138,14 +140,14 @@ class string
     /// @param [in] rhs is the copy origin
     ///
     /// @return reference to self
-    string& operator=(const string& rhs) noexcept;
+    string& operator=(const string& rhs) & noexcept;
```

Comment:
I think we have decided to disable this rule as well, so the `&` overload is not required.

Reply:
@saif-at-github The rule is now disabled.